### PR TITLE
Use NPM package ip to display ip

### PR DESCRIPTION
**Pull request by Justman10000**

### CODE REVIEW

**Comment by julianlam** [comment]
Hi, thanks for the contribution... May I ask why we'd want to use the dependent package?

**Comment by julianlam** [comment]
Keep in mind that "because Hacktoberfest" is not a valid reason 😆

**Comment by Justman10000** [comment]
> May I ask why we'd want to use the dependent package?

Is in commit! You can see it in the commit...

File: install/web.js [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
 const childProcess = require('child_process');
-const less = require('less');
 
-const webpack = require('webpack');
 const nconf = require('nconf');
+const webpack = require('webpack');
 
-const Benchpress = require('benchpressjs');
 const mkdirp = require('mkdirp');
+const Benchpress = require('benchpressjs');
```
Comment (julianlam) [inline install/web.js:16]: I do not see the point in re-ordering these dependencies

File: install/web.js [hunk 2]
```diff
@@ -95,7 +97,7 @@ async function runWebpack() {
 
 function launchExpress(port) {
 	server = app.listen(port, () => {
-		winston.info('Web installer listening on http://%s:%s', '0.0.0.0', port);
+		winston.info('Web installer listening on http://%s:%s', ip.address(), port);
```
Comment (julianlam) [inline install/web.js:100]: This is incorrect. `express` in the wen installer is binding to `0.0.0.0`, not the external IP address.

File: src/database/mongo.js [hunk 3]
```diff
@@ -2,11 +2,12 @@
 'use strict';
 
 
-const winston = require('winston');
+const ip = require('ip');
 const nconf = require('nconf');
 const semver = require('semver');
 const prompt = require('prompt');
 const utils = require('../utils');
+const winston = require('winston');
```
Comment (julianlam) [inline src/database/mongo.js:10]: Unnecessary re-ordering

File: src/database/mongo.js [hunk 4]
```diff
@@ -28,7 +29,7 @@ mongoModule.questions = [
 	{
 		name: 'mongo:host',
 		description: 'Host IP or address of your MongoDB instance',
-		default: nconf.get('mongo:host') || '127.0.0.1',
+		default: nconf.get('mongo:host') || ip.address(),
```
Comment (julianlam) [inline src/database/mongo.js:32]: The default should be to assume the database is installed on the local machine, and to use the shorthand `127.0.0.1`. If the default were the external IP of the server, and the installation was moved, the config would no longer be correct.

File: src/database/postgres.js [hunk 5]
```diff
@@ -14,7 +15,7 @@ postgresModule.questions = [
 	{
 		name: 'postgres:host',
 		description: 'Host IP or address of your PostgreSQL instance',
-		default: nconf.get('postgres:host') || '127.0.0.1',
+		default: nconf.get('postgres:host') || ip.address(),
```
Comment (julianlam) [inline src/database/postgres.js:18]: The default should be to assume the database is installed on the local machine, and to use the shorthand `127.0.0.1`. If the default were the external IP of the server, and the installation was moved, the config would no longer be correct.

File: src/database/redis.js [hunk 6]
```diff
@@ -12,7 +13,7 @@ redisModule.questions = [
 	{
 		name: 'redis:host',
 		description: 'Host IP or address of your Redis instance',
-		default: nconf.get('redis:host') || '127.0.0.1',
+		default: nconf.get('redis:host') || ip.address(),
```
Comment (julianlam) [inline src/database/redis.js:16]: The default should be to assume the database is installed on the local machine, and to use the shorthand `127.0.0.1`. If the default were the external IP of the server, and the installation was moved, the config would no longer be correct.

**Review by julianlam — CHANGES_REQUESTED** [review]
It seems unnecessary to use the `ip` module here.

**Comment by Justman10000** [comment]
@julianlam

What if it's not installed locally? But please, just destroy the work of your community, you won't see any more PRs from me!

**Comment by Justman10000** [comment]
It was meant to be able to copy the IP directly instead of having to replace 0.0.0.0 manually! And remember, more commits can always be added to PRs! It could have been fixed in other ways, but the main thing was to destroy work first...

**Comment by Justman10000** [comment]
Oh yeah, and I can't reach NodeBB on my VM by going to localhost on my actual PC as the VM has its own local IPv4!

**Comment by Justman10000** [comment]
Oh yeah, and I can't reach NodeBB on my VM by going to localhost on my really PC as the VM has its own local IPv4!

**Comment by Justman10000** [comment]
Oh yeah, and I can't reach NodeBB on my VM by typing localhost on my browser from my real PC as the VM has a own local IPv4!

**Comment by julianlam** [comment]
If you are installing nodeBB inside of a vm, then you should be replacing the IP address with the correct value on setup, and not enforcing the change to everybody else.

The pull request was closed not because additional commits could resolve the issues, but because every single line was unnecessary.

**Comment by Justman10000** [comment]
See #10947

**Pull request closed without merging**